pandas/core/groupby/groupby.py

```diff
@@ -1099,6 +1099,24 @@ def _wrap_transformed_output(self, output: Mapping[base.OutputKey, ArrayLike]):
     def _wrap_applied_output(self, data, keys, values, not_indexed_same: bool = False):
         raise AbstractMethodError(self)
 
+    def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
+        """
+        For SeriesGroupBy we want the default to be False (to match Series behavior).
+        For DataFrameGroupBy we want it to be True (for backwards-compat).
+        """
+        # GH#41291
+        if numeric_only is lib.no_default:
```

Comment:
are we deprecating the dataframe groupby behavior here?

Reply:
no, this is just a way of having the effective default be different for SeriesGroupBy vs DataFrameGroupBy